meson.build

```diff
@@ -103,7 +105,7 @@ if get_option('ipc')
     )
     nanoarrow_ipc_dep = declare_dependency(include_directories: [incdir],
                                            link_with: nanoarrow_ipc_lib,
-                                           dependencies: [nanoarrow_dep, flatcc_dep])
+                                           dependencies: [nanoarrow_dep])
```

Comment:
I think it was a mistake to include the `flatcc_dep` transitively here; I think its only required to build the `nanoarrow_ipc` lib itself, but shouldn't be required by libraries that want to use `nanoarrow_ipc` (?)

Reply:
Yea this removal here essentially removes the flatcc headers from being included by downstream targets that use the nanoarrow_ipc_dep; this is akin to changing the flatcc target inclusion from PUBLIC to PRIVATE in CMake